### ISSUE TITLE
Maint: numpy deprecation and future warnings

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#4228]
**update**

- rcond has been fixed, pinned to old behavior 
- issubdtype in tsa has been fixed 

still open:

- unicode in genfromtxt
- pandas and others (which were not listed here in the original description)
  see https://github.com/statsmodels/statsmodels/pull/4235#issuecomment-375391861

below is original

------------

see https://github.com/statsmodels/statsmodels/issues/3994#issuecomment-360829559 for changes in rcond in lstsq future warning

```
statsmodels/genmod/tests/test_glm.py::TestGlmBernoulli::()::test_params
  c:\py\lib\site-packages\numpy\lib\npyio.py:2258: VisibleDeprecationWarning: Reading unicode strings without specifying the encoding argument is deprecated. Set the encoding, use None for the system default.
    output = genfromtxt(fname, **kwargs)
statsmodels/genmod/tests/test_glm.py::test_perfect_pred
```

I thought we are reading binary not text ? Did the default change?

```
statsmodels/tsa/statespace/tests/test_sarimax.py::TestSARIMAXStatsmodels::()::test_loglike
  C:\projects\statsmodels\statsmodels\tsa\kalmanf\kalmanfilter.py:646: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
    if issubdtype(paramsdtype, float):
  C:\projects\statsmodels\statsmodels\tsa\kalmanf\kalmanfilter.py:650: FutureWarning: Conversion of the second argument of issubdtype from `complex` to `np.complexfloating` is deprecated. In future, it will be treated as `np.complex128 == np.dtype(complex).type`.
    elif issubdtype(paramsdtype, complex):
statsmodels/tsa/tests/test_arima.py::Test_Y_ARMA11_NoConst::()::test_params
  C:\projects\statsmodels\statsmodels\tsa\kalmanf\kalmanfilter.py:577: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
    if issubdtype(paramsdtype, float):
statsmodels/tsa/tests/test_arima.py::Test_ARIMA111::()::test_params
```

maybe more

another/several FutureWarning: from https://travis-ci.org/statsmodels/statsmodels/jobs/333445133
```
statsmodels/tsa/vector_ar/tests/test_vecm.py::test_log_like
  /home/travis/miniconda2/envs/statsmodels-test/lib/python3.5/site-packages/numpy/lib/shape_base.py:431: FutureWarning: in the future np.array_split will retain the shape of arrays with a zero size, instead of replacing them by `array([])`, which always has a shape of (0,).
    FutureWarning)
```

**Comment by josef-pkt** [comment]
@ChadFulton  Do you know what the `issubdtype` FutureWarning means?

**Comment by ChadFulton** [comment]
> On Fri, Jan 26, 2018 at 11:27 AM, Josef Perktold <notifications@github.com> wrote:
> @ChadFulton Do you know what the issubdtype FutureWarning means?
> 

I think this is related to https://docs.scipy.org/doc/numpy-1.14.0/release.html#future-changes and I think that it means that we need to explicitly use np.float64 (in issubdtype) if we actually mean a double, or else np.floating if we don't care if it's double or single, and similar for complex.

Chad

**Comment by josef-pkt** [comment]
@ChadFulton  Thanks, can you make the change?
AFAICS, the corresponding cython function in kalman_loglike use `double` and `dcomplex` as types. (But I never did anything beyond baby examples in cython.)

**Comment by ChadFulton** [comment]
Yes, no problem I'll hopefully get a chance tonight, I think it's a simple fix.

**Comment by josef-pkt** [comment]
matplotlib/numpy histogram
new keyword `density` in numpy >= 1.6, only relevant for unequal bin sizes
matplotlib adjusted in 2.1.x https://github.com/matplotlib/matplotlib/issues/9557/
in numpy "It will be removed in NumPy 2.0.0."

problem: either we ignore the 'normed' deprecation warning or we need to use normed/density conditional on numpy and matplotlib versions, as long as we use old versions of them.

histogram is in statespace `plot_diagnostics`

```
statsmodels/tsa/statespace/tests/test_sarimax.py::Test_ar_as_polynomial::()::test_results
  /home/travis/miniconda/envs/statsmodels-test/lib/python3.6/site-packages/matplotlib/axes/_axes.py:6462: UserWarning: The 'normed' kwarg is deprecated, and has been replaced by the 'density' kwarg.
    warnings.warn("The 'normed' kwarg is deprecated, and has been "
```

**Comment by josef-pkt** [comment]
genfromtxt problem, not investigated yet

e.g.
```
statsmodels/genmod/tests/test_glm.py::TestGlmInvgaussLog::()::test_params
  /home/travis/miniconda/envs/statsmodels-test/lib/python3.6/site-packages/numpy/lib/npyio.py:2266: VisibleDeprecationWarning: Reading unicode strings without specifying the encoding argument is deprecated. Set the encoding, use None for the system default.
    output = genfromtxt(fname, **kwargs)
```

**Comment by josef-pkt** [comment]
another one

The warning comes from numpy, but IIRC it's caused by pandas.
Needs checking, I don't remember any details (and I don't have the offending versions installed on my computers)

```
statsmodels/graphics/tests/test_boxplots.py::test_violinplot_beanplot
  /home/travis/miniconda/envs/statsmodels-test/lib/python3.6/site-packages/numpy/core/fromnumeric.py:52: FutureWarning: reshape is deprecated and will raise in a subsequent release. Please use .values.reshape(...) instead
    return getattr(obj, method)(*args, **kwds)
```

**Comment by josef-pkt** [comment]
and another one

pandas seems to have deprecated `argmax`

```
statsmodels/imputation/tests/test_ros.py::Test__ros_sort::()::test_censored_greater_than_max
  /home/travis/build/statsmodels/statsmodels/statsmodels/imputation/tests/test_ros.py:207: FutureWarning: 'argmax' is deprecated. Use 'idxmax' instead. The behavior of 'argmax' will be corrected to return the positional maximum in the future. Use 'series.values.argmax' to get the position of the maximum now.
    max_row = df['conc'].argmax()
```

**Comment by josef-pkt** [comment]
I'm not silencing the genfromtxt warning in test all datasets. This might require a more serious rewrite or deprecating load in favor of load_pandas, and I will open a new issue for 0.10.

Then, this issue can be closed when #4432 is merged.
